backend/inspirehep/search/factories/search.py

```diff
@@ -17,6 +17,12 @@
 
 
 def get_search_with_source(search):
+    request_fields = request.values.get("fields", "", type=str)
+    if request_fields:
```

Comment:
* what happens if you provide an alternative mime-type, or a `format` arg, together with `fields`? I think it should be a 400 (as it might screw up the other formats) if you request something different from JSON.
* what happens if you try to include a field that the user is not allowed to see? I suppose this will work correctly (due to filtering happening in the search response serializer at a later stage).

Reply:
I agree with the `400`, we better handle that.

Reply:
Good point, didn't think that the other formats would fail. 
About the hidden fields, it is handled at the serializer level so there isn't any issue.